### PR TITLE
Caffe

### PR DESCRIPTION
**Pull request by bvanessen**
Added the hash for the 1.0 release of caffe and put in a conflict
statemet indicating that protobuf requires a c++11 compiler.

### CODE REVIEW

File: var/spack/repos/builtin/packages/protobuf/package.py [hunk 1]
```diff
@@ -41,6 +41,8 @@ class Protobuf(AutotoolsPackage):
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
 
+    conflicts('%gcc@:4.8') # Requires c++11
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/protobuf/package.py]: Don't GCC 4.7 and 4.8 support C++11?
https://gcc.gnu.org/projects/cxx-status.html

**Comment by bvanessen** [comment]
They support part of the spec, but not all of it from what I can see.  I don’t have those compilers available to test, so i was conservative in bounding it to use at least 4.9.x

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470

**Comment by adamjstewart** [comment]
Which GCC version failed for you?

**Comment by bvanessen** [comment]
4.4.7



Brian Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) ‭925-290-5470‬
________________________________

**Comment by adamjstewart** [comment]
I would be conservative and still let people use GCC 4.7 and 4.8 unless we know otherwise.

File: var/spack/repos/builtin/packages/protobuf/package.py [hunk 2]
```diff
@@ -41,6 +41,8 @@ class Protobuf(AutotoolsPackage):
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
 
+    conflicts('%gcc@:4.4.7') # Requires c++11
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/protobuf/package.py]: I think `:4.6` would be better. We know that 4.6 doesn't accept `--std=c++11`, but 4.7 should.

File: var/spack/repos/builtin/packages/protobuf/package.py [hunk 3]
```diff
@@ -41,6 +41,8 @@ class Protobuf(AutotoolsPackage):
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
 
+    conflicts('%gcc@:4.6') # Requires c++11
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/protobuf/package.py]: This is the `flake8` test failing:
```
var/spack/repos/builtin/packages/protobuf/package.py:44: [E261] at least two spaces before inline comment
```

**Pull request merged**